Makefile

```diff
@@ -102,7 +107,7 @@ $(BIN_DIR)/azure-cloud-node-manager: $(PKG_CONFIG) $(wildcard cmd/cloud-node-man
 	CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} go build -a -o $(BIN_DIR)/azure-cloud-node-manager $(shell cat $(PKG_CONFIG)) ./cmd/cloud-node-manager
 
 $(BIN_DIR)/azure-cloud-node-manager.exe: $(PKG_CONFIG) $(wildcard cmd/cloud-node-manager/*) $(wildcard cmd/cloud-node-manager/**/*) $(wildcard pkg/**/*) ## Build node-manager binary for Windows.
-	CGO_ENABLED=0 GOOS=windows go build -a -o $(BIN_DIR)/azure-cloud-node-manager.exe $(shell cat $(PKG_CONFIG)) ./cmd/cloud-node-manager
```

Comment:
Makefile target should be `$(BIN_DIR)/azure-cloud-node-manager-${ARCH}.exe`. Or we don't need the `ARCH` suffix?

Reply:
I tested the target respectively by `WINDOWS_OSVERSION=1809 ARCH=arm64 make build-node-image-windows` with output

```
CGO_ENABLED=0 GOOS=windows GOARCH=arm64 go build -a -o bin/azure-cloud-node-manager-arm64.exe -ldflags '-s -w -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitVersion=v1.23.0-23-g2ef2d846d -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitCommit=2ef2d846d7ecb982a02d9ce82087ac09acf3badb -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitMajor=1 -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitMinor=23+ -X sigs.k8s.io/cloud-provider-azure/pkg/version.buildDate=2022-01-06T14:52:13Z' ./cmd/cloud-node-manager
```
and 

`WINDOWS_OSVERSION=1809 ARCH=amd64 make build-node-image-windows` with output

```
CGO_ENABLED=0 GOOS=windows GOARCH=amd64 go build -a -o bin/azure-cloud-node-manager-amd64.exe -ldflags '-s -w -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitVersion=v1.23.0-23-g2ef2d846d -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitCommit=2ef2d846d7ecb982a02d9ce82087ac09acf3badb -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitMajor=1 -X sigs.k8s.io/cloud-provider-azure/pkg/version.gitMinor=23+ -X sigs.k8s.io/cloud-provider-azure/pkg/version.buildDate=2022-01-06T14:52:13Z' ./cmd/cloud-node-manager
```

Reply:
What i mean is that `target` should be the name of the binary. (ref: https://www.gnu.org/software/make/manual/html_node/Rule-Introduction.html)

And as the linux version `$(BIN_DIR)/azure-cloud-node-manager` does not suffix with `ARCH`...

Reply:
The doc says usually.  Don't you think that will give more complexity to the target?

_A target is usually the name of a file that is generated by a program_